site/src/modules/resources/AgentDevcontainerCard.tsx

```diff
@@ -1,56 +1,101 @@
-import {
-	HelpTooltip,
-	HelpTooltipContent,
-	HelpTooltipText,
-	HelpTooltipTitle,
-	HelpTooltipTrigger,
-} from "components/HelpTooltip/HelpTooltip";
 import { Spinner } from "components/Spinner/Spinner";
+import { Stack } from "components/Stack/Stack";
 import {
 	Tooltip,
 	TooltipContent,
 	TooltipProvider,
 	TooltipTrigger,
 } from "components/Tooltip/Tooltip";
-import { ExternalLinkIcon } from "lucide-react";
+import { useProxy } from "contexts/ProxyContext";
+import { Container, ExternalLinkIcon } from "lucide-react";
+import { useFeatureVisibility } from "modules/dashboard/useFeatureVisibility";
+import { AppStatuses } from "pages/WorkspacePage/AppStatuses";
 import type { FC } from "react";
 import { useEffect, useState } from "react";
 import { portForwardURL } from "utils/portForward";
+import { Apps, organizeAgentApps } from "./AgentApps/AgentApps";
 import { AgentButton } from "./AgentButton";
-import { AgentDevcontainerSSHButton } from "./SSHButton/SSHButton";
+import { AgentLatency } from "./AgentLatency";
+import { SubAgentStatus } from "./AgentStatus";
+import { PortForwardButton } from "./PortForwardButton";
+import { AgentSSHButton } from "./SSHButton/SSHButton";
+import { SubAgentOutdatedTooltip } from "./SubAgentOutdatedTooltip";
 import { TerminalLink } from "./TerminalLink/TerminalLink";
 import { VSCodeDevContainerButton } from "./VSCodeDevContainerButton/VSCodeDevContainerButton";
 
 type AgentDevcontainerCardProps = {
-	agent: WorkspaceAgent;
-	container: WorkspaceAgentContainer;
+	parentAgent: WorkspaceAgent;
+	subAgents: WorkspaceAgent[];
+	devcontainer: WorkspaceAgentDevcontainer;
 	workspace: Workspace;
+	template: Template;
 	wildcardHostname: string;
 };
 
 export const AgentDevcontainerCard: FC<AgentDevcontainerCardProps> = ({
-	agent,
-	container,
+	parentAgent,
+	subAgents,
+	devcontainer,
 	workspace,
+	template,
 	wildcardHostname,
 }) => {
-	const folderPath = container.labels["devcontainer.local_folder"];
-	const containerFolder = container.volumes[folderPath];
-	const [isRecreating, setIsRecreating] = useState(false);
+	const { browser_only } = useFeatureVisibility();
+	const { proxy } = useProxy();
+
+	const [isRebuilding, setIsRebuilding] = useState(false);
+
+	// Track sub agent removal state to improve UX. This will not be needed once
+	// the devcontainer and agent responses  are aligned.
+	const [subAgentRemoved, setSubAgentRemoved] = useState(false);
+
+	// The sub agent comes from the workspace response whereas the devcontainer
+	// comes from the agent containers endpoint. We need alignment between the
+	// two, so if the sub agent is not present or the IDs do not match, we
+	// assume it has been removed.
+	const subAgent = subAgents.find((sub) => sub.id === devcontainer.agent?.id);
+
+	const appSections = (subAgent && organizeAgentApps(subAgent.apps)) || [];
+	const displayApps =
+		subAgent?.display_apps.filter((app) => {
+			switch (true) {
```

Comment:
It is not very common usage of switch, at least for JS developers. Can we replace it by a regular if/else?